### PR TITLE
Add TO Client API for ServerServerCapability Automation

### PR DESCRIPTION
**Pull request by dmohan001c**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Traffic Control Client 
- Traffic Ops
- CI tests

## What is the best way to verify this PR?
Execute all the Integration tests and make sure the tests passed.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/v4-client/server_server_capabilities.go [hunk 1]
```diff
@@ -39,7 +39,9 @@ func (to *Session) DeleteServerServerCapability(serverID int, serverCapability s
 	if opts.QueryParameters == nil {
 		opts.QueryParameters = url.Values{}
 	}
-	opts.QueryParameters.Set("serverId", strconv.Itoa(serverID))
+	if serverID != 0 {
+		opts.QueryParameters.Set("serverId", strconv.Itoa(serverID))
+	}
```
Comment (ocket8888) [inline traffic_ops/v4-client/server_server_capabilities.go:44]: This method doesn't work without specifying a server ID - why do this?

Comment (dmohan001c) [reply traffic_ops/v4-client/server_server_capabilities.go]: reverted the client method changes.

File: traffic_ops/testing/api/v4/serverservercapability_test.go [hunk 2]
```diff
@@ -36,14 +37,17 @@ func TestServerServerCapabilities(t *testing.T) {
...
 	})
 }
 
+/*
 func TestServerServerCapabilitiesForTopologies(t *testing.T) {
 	WithObjs(t, []TCObj{CDNs, Types, Tenants, Parameters, Profiles, Statuses, Divisions, Regions, PhysLocations, CacheGroups, Servers, Topologies, DeliveryServices, ServerCapabilities, ServerServerCapabilitiesForTopologies, TopologyBasedDeliveryServiceRequiredCapabilities}, func() {
 		DeleteTestServerServerCapabilitiesForTopologiesValidation(t)
 	})
-}
+}*/
```
[the remark this answers is not in the document]
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/serverservercapability_test.go]: it has been commented accidentally, I removed it.

File: traffic_ops/testing/api/v4/serverservercapability_test.go [hunk 3]
```diff
@@ -596,3 +597,99 @@ func GetDeliveryServiceServersWithCapabilities(t *testing.T) {
...
+	sscs, _, err := TOSession.GetServerServerCapabilities(client.RequestOptions{})
+	if err != nil {
+		t.Fatalf("cannot get server/Capability associations: %v - alerts: %+v", err, sscs.Alerts)
+	}
+	if len(sscs.Response) < 1 {
+		t.Fatalf("No Server Server Capability available to test invalid scenario")
+	}
+
+	//Delete Server Server Capability with Invalid Server Capability
+	alerts, _, err := TOSession.DeleteServerServerCapability(*sscs.Response[0].ServerID, "abcd", client.RequestOptions{})
```
[the remark this answers is not in the document]
Comment (dmohan001c) [reply traffic_ops/testing/api/v4/serverservercapability_test.go:689]: Fixed.

**Comment by rawlinp** [comment]
Looks like this PR had a bad merge because it's adding a bunch of already-existing code changes.

**Comment by dmohan001c** [comment]
> Looks like this PR had a bad merge because it's adding a bunch of already-existing code changes.

Fixed.

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**